src/main/java/com/alipay/oceanbase/rpc/table/ObTable.java

```diff
@@ -394,8 +391,28 @@ public ObPayload execute(final ObPayload request) throws RemotingException,
         } catch (Exception ex) {
             throw new ObTableConnectionStatusException("check status failed", ex);
         }
-
-        return realClient.invokeSync(connection, request, obTableExecuteTimeout);
+        boolean needReconnect = false;
+        int retryTimes = 0;
+        ObPayload payload = null;
+        do {
+            retryTimes++;
+            try {
+                if (needReconnect) {
+                    String msg = String.format("Receive error: tenant not in server and reconnect it, ip:{}, port:{}, tenant id:{}, retryTimes: {}",
+                            connection.getObTable().getIp(), connection.getObTable().getPort(), connection.getTenantId(), retryTimes);
+                    connection.reConnectAndLogin(msg);
```

Comment:
Will the connection be held by different threads, while other threads are accessing the server by the connection, we reconnect this connection. Which causes other threads to hang forever.

Reply:
Yes, maybe it will. I fix it through following way: force to close the connection if it is available before reconnecting it. When other threads hold this connection, they likely received the same -5150 error and close the connection and retry it will be the better way.